storm-core/src/jvm/org/apache/storm/container/cgroup/CgroupCommon.java

```diff
@@ -0,0 +1,247 @@
+            tasks.add(Integer.valueOf(task));
+        }
+        return tasks;
+    }
+
+    @Override
+    public void addProcs(int pid) throws IOException {
+        CgroupUtils.writeFileByLine(CgroupUtils.getDir(this.dir, CGROUP_PROCS), String.valueOf(pid));
+    }
+
+    @Override
+    public Set<Integer> getPids() throws IOException {
+        List<String> stringPids = CgroupUtils.readFileByLine(CgroupUtils.getDir(this.dir, CGROUP_PROCS));
+        Set<Integer> pids = new HashSet<Integer>();
+        for (String task : stringPids) {
+            pids.add(Integer.valueOf(task));
+        }
+        return pids;
+    }
+
+    @Override
+    public void setNotifyOnRelease(boolean flag) throws IOException {
+
+        CgroupUtils.writeFileByLine(CgroupUtils.getDir(this.dir, NOTIFY_ON_RELEASE), flag ? "1" : "0");
+    }
+
+    @Override
+    public boolean getNotifyOnRelease() throws IOException {
+        return CgroupUtils.readFileByLine(CgroupUtils.getDir(this.dir, NOTIFY_ON_RELEASE)).get(0).equals("1") ? true : false;
+    }
+
+    @Override
+    public void setReleaseAgent(String command) throws IOException {
+        if (!this.isRoot) {
+            return;
+        }
+        CgroupUtils.writeFileByLine(CgroupUtils.getDir(this.dir, RELEASE_AGENT), command);
+    }
+
+    @Override
+    public String getReleaseAgent() throws IOException {
+        if (!this.isRoot) {
+            return null;
+        }
+        return CgroupUtils.readFileByLine(CgroupUtils.getDir(this.dir, RELEASE_AGENT)).get(0);
+    }
+
+    @Override
+    public void setCgroupCloneChildren(boolean flag) throws IOException {
+        if (!getCores().keySet().contains(SubSystemType.cpuset)) {
+            return;
+        }
+        CgroupUtils.writeFileByLine(CgroupUtils.getDir(this.dir, CGROUP_CLONE_CHILDREN), flag ? "1" : "0");
+    }
+
+    @Override
+    public boolean getCgroupCloneChildren() throws IOException {
+        return CgroupUtils.readFileByLine(CgroupUtils.getDir(this.dir, CGROUP_CLONE_CHILDREN)).get(0).equals("1") ? true : false;
+    }
+
+    @Override
+    public void setEventControl(String eventFd, String controlFd, String... args) throws IOException {
+        StringBuilder sb = new StringBuilder();
+        sb.append(eventFd);
+        sb.append(' ');
+        sb.append(controlFd);
+        for (String arg : args) {
+            sb.append(' ');
+            sb.append(arg);
+        }
+        CgroupUtils.writeFileByLine(CgroupUtils.getDir(this.dir, CGROUP_EVENT_CONTROL), sb.toString());
+    }
+
+    public Hierarchy getHierarchy() {
+        return hierarchy;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public String getDir() {
+        return dir;
+    }
+
+    public CgroupCommon getParent() {
+        return parent;
+    }
+
+    public Set<CgroupCommon> getChildren() {
+
+        File file = new File(this.dir);
+        File[] files = file.listFiles();
+        if (files == null) {
```

Comment:
please look the function definition for listFiles().  The function can return null